commit 765354b50c2886fc0d294d6be3b207f7ae2ada70
Author: Jia Tan <jiat0218@gmail.com>
Date:   Wed Jan 11 01:18:50 2023 +0800

    Tests: Fix unused function warning in test_block_header.
    
    One of the global arrays of filters was only used in a test that
    required both encoders and decoders to be configured in the build.

diff --git a/tests/test_block_header.c b/tests/test_block_header.c
index 6b42df22..747925b3 100644
--- a/tests/test_block_header.c
+++ b/tests/test_block_header.c
@@ -35,6 +35,9 @@ static lzma_filter filters_one[2] = {
 };
 
 
+// These filters are only used in test_lzma_block_header_decode()
+// which only runs if encoders and decoders are configured.
+#ifdef HAVE_DECODERS
 static lzma_filter filters_four[5] = {
 	{
 		.id = LZMA_FILTER_X86,
@@ -52,6 +55,7 @@ static lzma_filter filters_four[5] = {
 		.id = LZMA_VLI_UNKNOWN,
 	}
 };
+#endif
 
 
 static lzma_filter filters_five[6] = {